Bali/src/main/java/OfferList.java

```diff
@@ -0,0 +1,117 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class OfferList {
+    private List<Offer> offers;
+
+    public OfferList(){
+        offers = new ArrayList<>();
+    }
+
+    public Offer getOffer(String id){
+        for(Offer offer : offers){
+            if(offer.getId().equals(id)){
+                return offer;
+            }
+        }
+        System.out.println("Incorrect type of id!");
+        return null;
+    }
+
+    public static boolean validateOffer(Offer offer){
+        if(offer.getId() != null && offer.getId().length() != 0 &&
+        offer.getDescription() != null && offer.getDescription().length() != 0 &&
+        offer.getCreatedAt() != null && offer.getCreatedAt().length() != 0 &&
+        offer.getLink() != null && offer.getLink().length() != 0 &&
+        offer.getVendor() != null && offer.getVendor().length() != 0 &&
+        offer.getHashTags() != null && offer.getHashTags().size() != 0 && offer.getHashTags().size() <= 7 &&
```

Comment:
Please avoid magic numbers. Can't remember what "7" is for, and no one can) this must be explained by the name of a constant)